src/bokeh/plotting/_geo_feature.py

```diff
@@ -0,0 +1,538 @@
+
+# External imports
+import numpy as np
+
+try:
+    import cartopy.crs as ccrs
+    import cartopy.feature as cfeature
+except ModuleNotFoundError as e:
+    raise ModuleNotFoundError(
+        "The module 'bokeh.plotting._geo_feature' requires the optional dependency 'cartopy'."
+        "To use this module, please install 'cartopy'.",
+    ) from e
+
+
+#-----------------------------------------------------------------------------
+# General API
+#-----------------------------------------------------------------------------
+
+GEO_FEATURE_COLORS = {
+    "LAND": "#EFEFDB",
+    "STATES": "#EFEFDB",
+    "RIVERS": "#9FDBF3",
+    "OCEAN": "#9FDBF3",
+    "LAKES": "#9FDBF3",
+    "BORDERS": "#CCCCCC",
+    "MAP_BOUNDARYS": "#000000",
+    "COASTLINES": "#000000",
+}
+
+def add_line_geometries(p, projection, geometries_collection, **line_kwargs):
+    """ Adds line geometries to a map with respect to a given projection.
+    The lines are added by the :func:`~bokeh.plotting.figure.multi_line` function.
+
+    Args:
+        p (Plot): Object which should be extended.
+        projection (cartopy.crs.Projection): Cartopy projection for a geographic map.
+        scale (str, "110m"): Scale of the feature resolution. Valid strings are "110m",
+            "50m" and "10m".
+
+    .. note::
+        This functions allows all keyword arguments defined by the
+        :func:`~bokeh.plotting.figure.multi_line` function.
+
+    Example:
+
+        .. bokeh-plot::
+            :source-position: above
+
+            import cartopy.crs as ccrs
+            import cartopy.feature as cfeature
+
+            from bokeh.plotting import figure, show
+            from bokeh.plotting._geo_feature import add_line_geometries
+
+            p = figure()
+            p = add_line_geometries(p, ccrs.PlateCarree(), cfeature.BORDERS)
+            show(p)
+    """
+    xs, ys = _collect_line_geometries(projection, geometries_collection)
+    p.multi_line(xs, ys, **line_kwargs)
+    return p
+
+
+def add_polygon_geometries(p, projection, geometries_collection, **poly_kwargs):
+    """ Adds polygon geometries to a map with respect to a given projection.
+    The polygons are added by the :func:`~bokeh.plotting.figure.multi_polygons`
+    function to allow holes and islands. To draw the border of the geometries,
+    set `draw_polygon_border` to `True`.
+
+    Args:
+        p (Plot): Object which should be extended.
+        projection (cartopy.crs.Projection): Cartopy projection for a geographic map.
+        scale (str, "110m"): Scale of the feature resolution. Valid strings are "110m",
+            "50m" and "10m".
+
+    Keyword Arguments:
+        draw_polygon_border (bool, False): Enables the plotting of the geometry border.
+        draw_polygon_color (str, "black"): Sets the color of the geometry border.
+
+            .. note::
+                This functions allows all keyword arguments defined by the
```

Comment:
```suggestion
                This functions allows all parameters and keyword arguments defined by the
```

Maybe I have to be more precise and should mention parameters and keyword arguments.

The `**line_kwargs` and `**poly_kwargs` notation is used to enable the full set of options. E.g. it is possible to add a legend item using `lagend_label`.

Reply:
The thing is, in the docs all the glyphs like `multi_ploygons` actually list all the kwargs explicitly

https://docs.bokeh.org/en/latest/docs/reference/plotting/figure.html#bokeh.plotting.figure.multi_polygons

I think just having `**poly_kwargs` with a short reference is not great for users. I suppose it would take some dedicated machinery like `glyph_method` decorator to automatically extract this information for better docs, though. Perhaps merge this sooner with a follow up issue about improving the docs.

Reply:
That machinery also makes, e.g. tab-complete work in IDEs so definitely a better UX

<img width="662" height="103" alt="Image" src="https://github.com/user-attachments/assets/f7369e9f-0435-48d0-8925-245e92e76bbb" />